### PR TITLE
Feature/user interface: Milestone 2 Designing the user interface

### PR DESCRIPTION
**Pull request by Cyrus-Kiprop**
Creation of the user interface  and an overall blueprint of the game logic

### CODE REVIEW

**Comment by shshamim63** [comment]
## Status :space_invader: Required Changes :space_invader: 
___
Hi @Cyrus-Kiprop & @Psiale :wave: ,
You have covered most of the corner cases of this milestone. I can see the messages for rendering current status of the board, congratulating the winner, asking user to choose a position from the valid options, also some comments that mentions about the game fate. Awesome job so far :clap: . But I have figured out some points that are pocking my mind to approve this milestone now. And I hope that you will agree with me. If no, don't hesitate to let me know. Lets get to those points now.
___
### :spades: Mandatory Changes :spades: 
___
1. game tells which player turn it is. All I can see is 
https://github.com/Cyrus-Kiprop/tic-tac-toe-OOP/blob/81f2fb798ec7dc46a750d94bb52fd83d056ecd85/bin/main#L54
https://github.com/Cyrus-Kiprop/tic-tac-toe-OOP/blob/81f2fb798ec7dc46a750d94bb52fd83d056ecd85/bin/main#L58
It seems confusing cause which player gonna place the move now. So for now you can add something like, `Player X(name) Please make a move....`
2. There are two possibilities in this game. Win or Draw. You have a message when a user is a winner. What about the Draw? I can not see that. Don't forget to add that too.
3. Game repeats all actions for next player's move. So that is a indication of using a loop in here. And a hint was provided in [OdinProject hint](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/OOP/milestone2/loop.png)
Lets divide this loop on several points. First of all it should iterate while the game is on. The game will quit if there is a winner/draw. Else both user will place their turn. Lets generate them in a short code format.
```ruby
game_on = true

# initial instructions for player, if you want to take username, symbol as input 

while game_on
  # loop for each move
  
  if winner # and/or draw
    game_on = false
  end
end
```
___
#### Please fix the above mentioned issue and send another review request.
___
## :herb: Happy Coding :herb: 
### Shakhawat Hossain
[Linkedin](https://www.linkedin.com/in/shakhawathossainshamim/)

File: bin/main [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+    end
+
+    if player == $players[1]['name']
+      puts "#{player} please choose you position in the board"
+      choose_position
+      puts '\n'
+      puts 'Position:'
+      gets.chomp
+      player = $players[1]['name']
+    end
```
Comment (stratospherique) [inline bin/main]: I think line 88 is preventing the turn switching mechanism.

- **[SUGGESTION]** you can avoid redundant code (i.e asking for player input) in a way that loop look as follows:
```ruby
    while game_on

      puts "#{player} please choose you position in the board"
      choose_position
      puts '\n'
      puts 'Position:'
      gets.chomp
      # here you should Check if the position is valid
      # check if the move give a winner/draw or not
       # if game not done
      player = player.token == 'X' ? $players[1] : $players[0] # To switch turn
      # else
      game_on = false
      # display results 
......
```

- You code either implement that in code, or just insert comments or write empty methods

**Review by stratospherique — CHANGES_REQUESTED** [review]
### Great work implementing the suggested changes :clap: However, further changes need to be applied in order for us to approve the milestone. Check :arrow_down: 

- **[REQUIRED]** Basic game flow is prepared: ([screenshot hint](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/OOP/milestone2/logic.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/59565018/))

* [x] `[REQUIRED]` game tells which player turn it is
* [x] `[REQUIRED]` game informs player if selected move is invalid
* [x] `[REQUIRED]` game informs player if selected move is a winning move
* [x] `[REQUIRED]` game informs player if selected move is a draw move
* [x] `[REQUIRED]` game repeats all actions for next player's move
- Check hints in code comments :arrow_double_down: 

### That's it :relaxed: When finished, kindly resubmit another code review request :pray: 

Happy coding :v: 
**_Ahmed Mahfoudh_**

**Review by fatymahmed — APPROVED** [review]
### Hi @Cyrus-Kiprop  @Psiale,
### Great job making the suggested changes🎉🎉
- Before merging, please change your `main` file to `main.rb`
### It's a 👍🏾 from me!
## ✔️ Approved!

Good luck with your next projects.
 _Don't forget to submit your project completion form_

**Pull request closed without merging**